### PR TITLE
Fixes #30943 - adds additional remote db notes

### PR DESCRIPTION
**Pull request by jjeffers**

### CODE REVIEW

File: plugins/katello/nightly/user_guide/remote_databases/index.md
Comment (jjeffers) [inline plugins/katello/nightly/user_guide/remote_databases/index.md]: @ianballou I would like to establish necessary messaging here, and propagate those backwards into each of the extant version. That way, any edits or changes we'd only have to change here, and then when final, add those messages to the other versions.

File: plugins/katello/3.15/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/3.15/user_guide/remote_databases/index.md]: The correct way to write it is PostgreSQL. I'd also use Pulp.

```suggestion
### Important note: Katello 3.15 changes the default database for Pulp from PostgreSQL v9.x to PostgreSQL v12. External databases should be upgraded to avoid conflicts.
```

It may also be good to note that Foreman itself recommends upgrading to version 12. https://theforeman.org/manuals/2.0/index.html#Headlinefeatures already mentions it. https://theforeman.org/manuals/2.0/index.html#3.1.1SupportedPlatforms also states you must at least run PostgreSQL 10 for Foreman itself.

This also suggests MongoDB is no longer needed for Pulp, but that's not really true.

File: plugins/katello/nightly/user_guide/remote_databases/index.md
Comment (ianballou) [inline plugins/katello/nightly/user_guide/remote_databases/index.md]: The entire text block being a larger font looks a little strange.  Maybe instead there could be an "Important notes" header with the text as normal below?

File: plugins/katello/nightly/user_guide/remote_databases/index.md
Comment (ianballou) [inline plugins/katello/nightly/user_guide/remote_databases/index.md]: "Pulpcore database related"?

**Review by ianballou — COMMENTED** [review]
Do you think we should also update the "Fresh install" and "Migration of existing Foreman" sections to include the Pulpcore settings? I think less-experienced users might still need to ask us for help since we don't lay the steps out for them.

File: plugins/katello/nightly/user_guide/remote_databases/index.md
Comment (jjeffers) [reply plugins/katello/nightly/user_guide/remote_databases/index.md]: @ianballou how about:
Screenshot from 2020-10-19 08-57-26

File: plugins/katello/3.15/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/3.15/user_guide/remote_databases/index.md]: One thing that a lot of 3.15 users ran into was that Pulpcore doesn't support external databases in this version, so you'll always end up with a local database. IMHO it was a mistake to release Pulpcore without external DBs and it would have saved a lot of support questions if we at least documented this.

File: plugins/katello/3.16/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/3.16/user_guide/remote_databases/index.md]: Wasn't 3.15 already based on Foreman 2.0?
```suggestion
Note that [Foreman v2.0 requires PostgreSQL v12](https://theforeman.org/manuals/2.0/index.html#Headlinefeatures).
```

Edit: Also, Foreman requires 10+, not 12. However, we only build `rh-postgresql12-postgresql-evr` for 12.

File: plugins/katello/3.15/user_guide/remote_databases/index.md
Comment (jjeffers) [reply plugins/katello/3.15/user_guide/remote_databases/index.md]: In that case maybe we should remove "External databases..." and instead point out that 3.15 doesn't support external databases?

Comment (ekohl) [reply plugins/katello/3.15/user_guide/remote_databases/index.md]: You can still use Foreman and Candlepin external databases. The Foreman DB must be updated to at least 10 on Foreman 2.0.

File: plugins/katello/3.15/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/3.15/user_guide/remote_databases/index.md]: :-1: on "changes the default database for Pulp". It gives the impression MongoDB is removed, but it's not. However, it's safe to to say Foreman 2.0 has dropped support for PostgreSQL 9.x and users should upgrade to 12.

I also strongly insist that a separate note is made that Pulp 3 is installed alongside, but that lacks remote database support. Having that would have saved me from answers the same question in support.

**Comment by ekohl** [comment]
Linking for completeness: https://community.theforeman.org/t/upgrade-2-0-2-3-15-3-2-1-2-3-16-0-db-migrate-fails-loading-evr-extension-when-using-external-postgresql-database-for-foreman-and-candlepin/20515/9 is a post by a community member with suggestions for improvements in the docs.

File: plugins/katello/3.15/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/3.15/user_guide/remote_databases/index.md]:
```suggestion
Pulpcore installs a separate PostgreSQL database and lacks support for external databases. This support is introduced in Katello 3.16.

Foreman and Pulpcore require PostgreSQL 10 or newer. Databases should be upgraded to avoid conflicts. It is recommended to update to version 12.
```

File: plugins/katello/3.16/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/3.16/user_guide/remote_databases/index.md]: Version 3.16 does support external databases
```suggestion
```

File: plugins/katello/3.16/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/3.16/user_guide/remote_databases/index.md]: Katello 3.16 runs on Foreman 2.1. This note belongs in the 3.15 docs.

File: plugins/katello/3.16/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/3.16/user_guide/remote_databases/index.md]:
```suggestion
2. Katello 3.16 adds a dependency on [postgresql-evr](https://github.com/katello/postgresql-evr). The Katello repository provides the `rh-postgresql12-postgresql-evr` package on EL7. This should be installed on any external PostgreSQL v12 database before applying any db migrations.
```

**Review by ekohl — CHANGES_REQUESTED** [review]
I didn't comment on 3.17 and nightly, but the 3.16 notes apply for those as well.

File: plugins/katello/3.16/user_guide/remote_databases/index.md
Comment (ekohl) [reply plugins/katello/3.16/user_guide/remote_databases/index.md]: This was marked as resolved, but 3.16 does support a separate PG DB for Pulpcore. If you keep a list, I'd suggest not using numbers but rather bullets.

File: plugins/katello/3.17/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/3.17/user_guide/remote_databases/index.md]: Should it be past tense here (`added`) indicating it's not a new requirement?

**Comment by jjeffers** [comment]
@ekohl @ianballou are there other items you would like to see added or amended in this PR? I think I got all the changes into the versions from nightly backwards to 3.15.

**Comment by ianballou** [comment]
@jjeffers it's looking fine to me.  I'll defer to @ekohl for the final approval.

File: plugins/katello/3.16/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/3.16/user_guide/remote_databases/index.md]: It was part of version 2.0, but it's not a headline feature in 2.1. The link should be removed but I'd also consider dropping it altogether. The evr package is only built for PG 12 so that already narrows it down for most users.

File: plugins/katello/3.15/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/3.15/user_guide/remote_databases/index.md]: In other files it uses bullet points, here it doesn't. Feels a bit inconsistent.

File: plugins/katello/3.15/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/3.15/user_guide/remote_databases/index.md]: You can drop this since it's duplicate from the line above. Perhaps you could add the link there.

**Review by ekohl — COMMENTED** [review]
Notes for 3.16 also go for newer versions. Not sure if 3.18 already branched. If so, it also needs to be updated.

File: plugins/katello/3.15/user_guide/remote_databases/index.md
Comment (jjeffers) [reply plugins/katello/3.15/user_guide/remote_databases/index.md]: @ekohl After I remove the second line, we no longer have "important notes" but an "important note: I wonder if that formatting should change too.

File: plugins/katello/3.15/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/3.15/user_guide/remote_databases/index.md]:
```suggestion
* Foreman ([since 2.0](https://theforeman.org/manuals/2.0/index.html#Headlinefeatures)) and Pulpcore require PostgreSQL 10+. Databases should be upgraded to avoid conflicts. It is recommended to update to version 12.
```

File: plugins/katello/3.15/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/3.15/user_guide/remote_databases/index.md]:
```suggestion
* Pulpcore installs a separate PostgreSQL database. The installer lacks support for external databases. This support is introduced in Katello 3.16.
```

File: plugins/katello/3.16/user_guide/remote_databases/index.md
Comment (ekohl) [reply plugins/katello/3.16/user_guide/remote_databases/index.md]: The link is still incorrect.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: plugins/katello/nightly/user_guide/remote_databases/index.md
Comment (ianballou) [inline plugins/katello/nightly/user_guide/remote_databases/index.md]: Typo: "upagrade"

Will the postgresql12-postgresql-evr note also be added to the 3.18, 3.17, and 3.16 docs pages?

Comment (jjeffers) [reply plugins/katello/nightly/user_guide/remote_databases/index.md]: @ianballou thank you for spotting that typo. I believe we had evr notes for previous version, but decided to remove them.

File: plugins/katello/3.18/user_guide/remote_databases/index.md
Comment (ianballou) [inline plugins/katello/3.18/user_guide/remote_databases/index.md:31]: I think we need the "Pulpcore database related:" section below here for 3.18.

After that hopefully we can merge?

Comment (jjeffers) [reply plugins/katello/3.18/user_guide/remote_databases/index.md:31]: @ianballou I will get to adding that asap.

Comment (ianballou) [reply plugins/katello/3.18/user_guide/remote_databases/index.md:31]: Cool. We'll need someone with merge access to give it a final look over. Maybe @ekohl since he was already in here.

File: plugins/katello/3.15/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/3.15/user_guide/remote_databases/index.md]: Not a native speaker, but isn't it plural?
```suggestion
* Foreman ([since 2.0])(https://theforeman.org/manuals/2.0/index.html#Headlinefeatures) and Pulpcore require PostgresSQL 10+. Databases should be upgraded to avoid conflicts. It is recommended to update to version 12.
```

File: plugins/katello/nightly/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/nightly/user_guide/remote_databases/index.md]: I think it's safe to say
```suggestion
Note that PostgreSQL v10+ is required and users are encouraged to upgrade to PostgreSQL v12.
```

AFAIK Pulpcore also needs version 9.6 or 10.

File: plugins/katello/nightly/user_guide/remote_databases/index.md
Comment (ekohl) [inline plugins/katello/nightly/user_guide/remote_databases/index.md]: As a user I'm not sure what `Databases should be upgraded to avoid conflicts.` means if it follows `Pulpcore installs a separate PostgreSQL database`. Upgraded to what?

**Review by ekohl — COMMENTED** [review]
Apologies for the slow review. It got lost in my backlog.

File: plugins/katello/3.15/user_guide/remote_databases/index.md
Comment (jjeffers) [reply plugins/katello/3.15/user_guide/remote_databases/index.md]: You are correct!

File: plugins/katello/nightly/user_guide/remote_databases/index.md
Comment (jjeffers) [reply plugins/katello/nightly/user_guide/remote_databases/index.md]: Good point re pulpcore.

**Review by jlsherrill — APPROVED** [review]
ACK, lets get this in for 4.0

**Pull request merged**
